src/reset-password/ResetPasswordPage.jsx

```diff
@@ -30,9 +32,12 @@ const ResetPasswordPage = (props) => {
   const [passwordMatch, setPasswordMatchValue] = useState(true);
   const [validationMessage, setvalidationMessage] = useState('');
   const [bannerErrorMessage, setbannerErrorMessage] = useState('');
+  const [apiFailureHeader, setApiFailureHeader] = useState('');
 
   useEffect(() => {
-    if (props.status === 'failure' && props.errors) {
+    if (props.status === 'failure' && props.errors === INTERNAL_SERVER_ERROR) {
+      setApiFailureHeader(intl.formatMessage(messages['reset.password.request.server.error']));
```

Comment:
Is this necessary to use a state for this message? It will rerender the component upon updating the state which could be avoided by directly passing this header message to `APIFailureMessage` component because I don't see any other scenario where we are using a different message for it.